iceoryx_posh/source/capro/service_description.cpp

```diff
@@ -73,31 +73,26 @@ ServiceDescription::ServiceDescription(const cxx::Serialization& serial) noexcep
 {
     std::underlying_type<Scope>::type scope = 0;
     std::underlying_type<Interfaces>::type interfaceSource = 0;
-    serial.extract(m_serviceString,
-                   m_instanceString,
-                   m_eventString,
-                   m_classHash[0u],
-                   m_classHash[1u],
-                   m_classHash[2u],
-                   m_classHash[3u],
-                   scope,
-                   interfaceSource);
-    if (scope > static_cast<std::underlying_type<Scope>::type>(Scope::INVALID))
+    auto deserializationSuccessful = serial.extract(m_serviceString,
```

Comment:
I guess this has to do with my comment regarding the invalid `ServiceDescription` from the previous PR? If yes, could you please refactor this further to something like this
```cpp
cxx::expected<ServiceDescription, cxx::Serialization::Error>
ServiceDescription::deserialize(const cxx::Serialization& serialized) noexcept;
```
so that the caller gets informed when this fails?

Reply:
Oh I totally overlooked this because I did not look in which method this is performed :( ...

@elBoberido you are right and shouldn't it be a static factory method of `ServiceDescription`?

Just as inspiration, in every class in which the CTor can fail we have the static factory method `create`. I would not use the creation pattern here (it should be slowly removed) but I would name the method `create` to be consistent with all the other occurrences all over the code.

Then we can make a rule. CTor succeeds always, if the CTor can fail we implement a static factory method called `create`.

Reply:
@elfenpiff ahem, you just have to look closely to spot the `static`. If you can't see it, just move your eyes closer to the screen. If you still can't see it, you might need some glasses ;)

Regarding the naming. As rule of thumb I would agree, but depending on the context it should be possible to deviate from this naming. There is for example `Duration::from_seconds` which deviates and makes the code better to understand. Similarly I would also suggest to call this `deserialize` since it has a special meaning. This is maybe more important if there is a `create` function because of the regular ctor could fail.

Reply:
Done